src/integrationTest/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/component/HomeOfficeStatusSearchIntegrationTest.java

```diff
@@ -0,0 +1,223 @@
+        addUserDetailsStub(server);
+        addHomeOfficeApiSearchNoInvolvementsStub(server, homeOfficeReference);
+        addHomeOfficeApiInstructStub(server, homeOfficeReference);
+
+        PreSubmitCallbackResponseForTest response = getPreSubmitCallbackResponse(homeOfficeReference);
+
+        AsylumCase asylumCase = response.getAsylumCase();
+
+        assertNotNull(response);
+        assertNotNull(asylumCase);
+        assertEquals(asylumCase.read(HOME_OFFICE_SEARCH_STATUS, String.class), Optional.of("FAIL"));
+        final Optional<HomeOfficeCaseStatus> homeOfficeCaseStatus =
+            asylumCase.read(HOME_OFFICE_CASE_STATUS_DATA, HomeOfficeCaseStatus.class);
+        assertFalse(homeOfficeCaseStatus.isPresent());
+    }
+
+    @Test
+    @WithMockUser(authorities = {"caseworker-ia-legalrep-solicitor"})
+    public void shouldHandleWhenExtraFieldsAreSent(@WiremockResolver
+        .Wiremock(factory = StaticPortWiremockFactory.class) WireMockServer server) throws Exception {
+
+        final String homeOfficeReference = "extra-fields-ref-number";
+
+        addServiceAuthStub(server);
+        addUserDetailsStub(server);
+        addHomeOfficeApiSearchExtraFieldsStub(server, homeOfficeReference);
+        addHomeOfficeApiInstructStub(server, homeOfficeReference);
+
+        IaCaseHomeOfficeIntegrationApiClient iaCaseHomeOfficeIntegrationApiClient =
+            new IaCaseHomeOfficeIntegrationApiClient(mockMvc);
+
+        final CallbackForTestBuilder callback = callback()
+            .event(Event.SUBMIT_APPEAL)
+            .caseDetails(someCaseDetailsWith()
+                .state(State.APPEAL_SUBMITTED)
+                .caseData(anAsylumCase()
+                    .with(AsylumCaseDefinition.APPEAL_REFERENCE_NUMBER, APPEAL_REFERENCE_NUMBER)
+                    .with(HOME_OFFICE_REFERENCE_NUMBER, homeOfficeReference)));
+
+
+        PreSubmitCallbackResponseForTest response = iaCaseHomeOfficeIntegrationApiClient.aboutToSubmit(callback);
+
+        AsylumCase asylumCase = response.getAsylumCase();
+
+        assertNotNull(response);
+        assertNotNull(asylumCase);
+        assertEquals(asylumCase.read(HOME_OFFICE_SEARCH_STATUS, String.class), Optional.of("SUCCESS"));
+        final Optional<HomeOfficeCaseStatus> homeOfficeCaseStatus =
+            asylumCase.read(HOME_OFFICE_CASE_STATUS_DATA, HomeOfficeCaseStatus.class);
+        Person person = null;
+        if (homeOfficeCaseStatus.isPresent()) {
+            person = homeOfficeCaseStatus.get().getPerson();
+        }
+        assertNotNull(person);
+        assertEquals(person.getGivenName(), "Capability");
+        assertEquals(person.getFamilyName(), "Smith");
+        assertEquals(person.getNationality().getDescription(), "Canada");
+        assertEquals(person.getFullName(), "Capability Smith");
+    }
+
+
+    @Test
+    @WithMockUser(authorities = {"caseworker-ia-legalrep-solicitor"})
+    public void shouldHandle400InternalServerError(@WiremockResolver
```

Comment:
4xx errors tend to be client errors rather than internal server errros.  Maybe rename?